subprojects/functional/src/main/java/org/gradle/internal/Try.java

```diff
@@ -221,7 +223,19 @@ public boolean isSuccessful() {
 
         @Override
         public T get() {
-            throw UncheckedException.throwAsUncheckedException(failure);
+            if (failure instanceof InterruptedException) {
```

Comment:
Ideally we'd have a single implementation of this logic but I don't have a strong feeling against duplicating this at this point in the time. I would keep a _TODO: consider updating <original site/inlined site> when changing this_ in both sites.

Reply:
I am fine with duplicating this as well, as long as we include the comment on both sites.

Reply:
Added.